whirlpool/auth.py

```diff
@@ -96,7 +96,8 @@ async def _do_auth(self, refresh_token):
                     elif refresh_token:
                         return await self._do_auth(refresh_token=None)
                     return None
-
+        except (aiohttp.ClientConnectionError, asyncio.TimeoutError) as ex:
+            raise ex
```

Comment:
The primary point is to let HA know there was a connect error or timeout so that it can try again later.  The original code path (after I found the error in the integration) will result in an invalid password declaration (which is not right).

Reply:
Maybe we could just do `async with aiohttp.ClientSession() as session:` after line 89, to get the session instance, and then we don't need the `finally` block at all. That means we can just remove the try/except/finally, and every exception will bubble up to HA.
What do you think?

Reply:
Looking over the requests the devs made for AladdinConnect, we are actually supposed to get  the session from HA with async_get_clientsession() and pass into the API, then use that session, and never close it in the API - HA takes care of it.

https://github.com/home-assistant/core/blob/8345176ccc1ca17e5b12960be13447e14fe23544/homeassistant/components/aladdin_connect/__init__.py#L27-L28

https://github.com/mkmer/AIOAladdinConnect/blob/e61607f56bbc9bedb83066c54e8f77cf4fb63e12/AIOAladdinConnect/__init__.py#L55

That would allow us to do the same, without finally, and the session would be passed in....
Thoughts?

Reply:
Yeah, that is a better approach!

Reply:
I think I got it setup with the "external" client session - appears to work with the cli tests.  Probably a mistake or two in there :(